### DISCUSSION TITLE
trim-newlines dependency of @quasar/icongenie vulnerability and UnhandledPromiseRejectionWarning

### DISCUSSION THREAD

**Discussion by vahost** [quasarframework/quasar#10586] [General - Components / Directives / etc]
These two issues don't seem to be connected, but I can't get rid of them, so it makes me wonder if they are somehow connected after all.

I am getting a white screen in my quasar app, so I updated node, npm, and quasar

When running npm install, I encounter this:

High            Regular Expression Denial of Service                          

  Package         trim-newlines                                                 

  Patched in      >=3.0.1 <4.0.0 || >=4.0.1                                     
  Dependency of   @quasar/icongenie                                             

  Path            @quasar/icongenie > imagemin-pngquant > pngquant-bin >        
                  logalot > squeak > lpad-align > meow > trim-newlines          

  More info       https://npmjs.com/advisories/1753   

I tried updating trim-newlines to @4.0.2, deleting package-lock.json, then re-running npm install, but the error won't go away. 

So I tried running quasar dev, and I get this error:

(node:17392) UnhandledPromiseRejectionWarning:   TypeError: Merging undefined is not supported

Software versions:
Node: v14.17.6
npm: 6.14.15
quasar: 3.1.0

The project was originally created with quasar 2.0.

**Comment by hawkeye64** [comment]
That is 7 upstream dependencies that need updating, starting with `trim-newlines` and then `meow` and so on before it'll work for icongenie.

Luckily, it's a runtime issue, so doesn't affect the output of icongenie or your project.

**Reply by vahost** [reply]
I've updated the dependencies, and those are now clearing okay when I run npm install.

Quasar dev also compiles without errors.

However, I'm still getting the white screen in the browser.

In Inspect console it reads:

client-entry.js?2f39:105 [Quasar] boot error: TypeError: Cannot read properties of undefined (reading '0')
    at _callee$ (xmlJs.js?08e1:26)
    at tryCatch (runtime.js?ad1f:63)
    at Generator.invoke [as _invoke] (runtime.js?ad1f:294)
    at Generator.eval [as next] (runtime.js?ad1f:119)
    at asyncGeneratorStep (asyncToGenerator.js?c973:3)
    at _next (asyncToGenerator.js?c973:25)
_callee$ @ client-entry.js?2f39:105

Any ideas about that?

**Comment by hawkeye64** [comment]
When you force patches, you never know what inconsistencies will be introduced. It's best to just wait them out and update as upstream packages update their own dependencies.
